src/neural/cuda/common_kernels.cu

```diff
@@ -466,6 +466,36 @@ void globalScale(int N, int C, T* output, const T* input, const T* scaleBias,
   ReportCUDAErrors(cudaGetLastError());
 }
 
+template <typename T>
+__global__ void policyMap_kernel(T* output, const T* input, const short* indices,
+                                 int N, int inputSize, int usedSize, int outputSize) {
+  int tid = blockIdx.x * blockDim.x + threadIdx.x;
+
+  int n = tid / usedSize;
+  int i = tid % usedSize;
+
+  if (n >= N) return;
+
+  int j = indices[i];
+
+  if (j >= 0) {
+    output[n * outputSize + j] = input[n * inputSize + i];
+  }
+}
+
+template <typename T>
+void PolicyMap(int N, T* output, const T* input, const short* indices,
+        int inputSize, int usedSize, int outputSize) {
```

Comment:
Formatting in this file doesn't look like its had the auto formatter run over it (possibly because of file extension).